code/game/gamemodes/miniantags/revenant/revenant.dm

```diff
@@ -344,20 +345,21 @@
 	icon = 'icons/effects/effects.dmi'
 	icon_state = "revenantEctoplasm"
 	w_class = WEIGHT_CLASS_SMALL
-	var/reforming = 1
+	var/reforming = FALSE // so spawning this doesnt instantly make a rev out of nowhere.
 	var/essence = 75 //the maximum essence of the reforming revenant
-	var/inert = 0
+	var/inert = FALSE
 	var/client/client_to_revive
 
 /obj/item/ectoplasm/revenant/New()
 	..()
-	reforming = 0
-	spawn(600) //1 minutes
-		if(src && reforming)
-			reform()
-		else
-			inert = 1
-			visible_message("<span class='warning'>[src] settles down and seems lifeless.</span>")
+	addtimer(CALLBACK(src, .proc/attempt_revive), 1 MINUTES)
+
+/obj/item/ectoplasm/revenant/proc/attempt_revive()
+	if(src && reforming)
```

Comment:
No need for the `src` check here. It was used in the spawn since spawns can have their `src` object be nulled.
A callback using a timer can never have a `src` that is null. Else the timer just won't go off

Reply:
Done!